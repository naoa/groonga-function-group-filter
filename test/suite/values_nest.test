plugin_register functions/group_filter
table_create prefectures TABLE_HASH_KEY ShortText
table_create applicants TABLE_HASH_KEY ShortText
table_create ipcs TABLE_HASH_KEY ShortText
table_create docs TABLE_HASH_KEY ShortText
column_create docs applicants COLUMN_VECTOR applicants
column_create docs ipcs COLUMN_VECTOR ShortText
load --table docs
[
{"_key": "JP2001213456", "applicants": ["三洋", "三洋オプテックデザイン"], "ipcs": ["G06F", "G04A"]},
{"_key": "JP2001213457", "applicants": ["パナソニック"], "ipcs": ["G04A"]},
{"_key": "JP2001213458", "applicants": ["ソニー"], "ipcs": ["G06F", "G04B"]},
{"_key": "JP2001213459", "applicants": ["パナソニック", "三洋電機"], "ipcs": ["G06F"]}
]
column_create applicants applicants COLUMN_INDEX docs applicants
column_create ipcs ipcs COLUMN_INDEX docs ipcs

column_create applicants prefecture COLUMN_SCALAR prefectures

load --table applicants
[
{"_key": "三洋オプテックデザイン", "prefecture": "岐阜"},
{"_key": "パナソニック", "prefecture": "大阪"},
{"_key": "ソニー", "prefecture": "東京"}
]

column_create prefectures applicants_index COLUMN_INDEX applicants prefecture

select docs \
  --filter 'values_filter("applicants.prefecture", "大阪 岐阜")' \
  --output_columns _id,_score,_key,applicants,#group_applicants_prefecture
