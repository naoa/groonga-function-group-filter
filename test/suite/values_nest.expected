plugin_register functions/group_filter
[[0,0.0,0.0],true]
table_create prefectures TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
table_create applicants TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
table_create ipcs TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
table_create docs TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
column_create docs applicants COLUMN_VECTOR applicants
[[0,0.0,0.0],true]
column_create docs ipcs COLUMN_VECTOR ShortText
[[0,0.0,0.0],true]
load --table docs
[
{"_key": "JP2001213456", "applicants": ["三洋", "三洋オプテックデザイン"], "ipcs": ["G06F", "G04A"]},
{"_key": "JP2001213457", "applicants": ["パナソニック"], "ipcs": ["G04A"]},
{"_key": "JP2001213458", "applicants": ["ソニー"], "ipcs": ["G06F", "G04B"]},
{"_key": "JP2001213459", "applicants": ["パナソニック", "三洋電機"], "ipcs": ["G06F"]}
]
[[0,0.0,0.0],4]
column_create applicants applicants COLUMN_INDEX docs applicants
[[0,0.0,0.0],true]
column_create ipcs ipcs COLUMN_INDEX docs ipcs
[[0,0.0,0.0],true]
column_create applicants prefecture COLUMN_SCALAR prefectures
[[0,0.0,0.0],true]
load --table applicants
[
{"_key": "三洋オプテックデザイン", "prefecture": "岐阜"},
{"_key": "パナソニック", "prefecture": "大阪"},
{"_key": "ソニー", "prefecture": "東京"}
]
[[0,0.0,0.0],3]
column_create prefectures applicants_index COLUMN_INDEX applicants prefecture
[[0,0.0,0.0],true]
select docs   --filter 'values_filter("applicants.prefecture", "大阪 岐阜")'   --output_columns _id,_score,_key,applicants,#group_applicants_prefecture
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        3
      ],
      [
        [
          "_id",
          "UInt32"
        ],
        [
          "_score",
          "Int32"
        ],
        [
          "_key",
          "ShortText"
        ],
        [
          "applicants",
          "applicants"
        ],
        [
          "#group_applicants_prefecture",
          "prefectures"
        ]
      ],
      [
        1,
        1,
        "JP2001213456",
        [
          "三洋",
          "三洋オプテックデザイン"
        ],
        [
          "岐阜"
        ]
      ],
      [
        2,
        1,
        "JP2001213457",
        [
          "パナソニック"
        ],
        [
          "大阪"
        ]
      ],
      [
        4,
        1,
        "JP2001213459",
        [
          "パナソニック",
          "三洋電機"
        ],
        [
          "大阪"
        ]
      ]
    ]
  ]
]
